File: backend/Models/SupportTicket.cs
using System.ComponentModel.DataAnnotations;

namespace Travyle.Api.Models;

public enum TicketPriority
{
    Low,
    Medium,
    High,
    Critical
}

public enum TicketStatus
{
    Pending_AI_Triage,
    Pending_Admin_Voucher_Approval,
    In_Review,
    Resolved,
    Closed,
    Rejected
}

public class SupportTicket
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    public Guid UserId { get; set; }
    public User? User { get; set; }

    public Guid? BookingId { get; set; }
    public Guid? TourId { get; set; }

    [Required]
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [Required]
    public string Description { get; set; } = string.Empty;

    [MaxLength(50)]
    public string Category { get; set; } = "General"; // TourDelay, TourQuality, Safety, Billing, GuideConduct, General

    public TicketPriority Priority { get; set; } = TicketPriority.Medium;

    public TicketStatus Status { get; set; } = TicketStatus.Pending_AI_Triage;

    public string? AttachmentUrl { get; set; }

    // AI Triage & Analysis attributes
    public double SentimentScore { get; set; } = 0.0; // -1.0 (very negative) to +1.0 (very positive)
    public string SeverityTier { get; set; } = "Tier_1_Low"; // Tier_1_Low, Tier_2_High, Tier_3_Critical
    public string? AiReasoning { get; set; }
    public string? ResolutionSummary { get; set; }

    public Guid? AssignedToAdminId { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public ICollection<AuditLog> AuditLogs { get; set; } = new List<AuditLog>();
    public ICollection<Voucher> Vouchers { get; set; } = new List<Voucher>();
}

File: mobile/lib/models/support_ticket.dart
class AuditLogModel {
  final String id;
  final String? supportTicketId;
  final String action;
  final String actorRole;
  final String? actorId;
  final String details;
  final String? metadataJson;
  final DateTime timestamp;

  AuditLogModel({
    required this.id,
    this.supportTicketId,
    required this.action,
    required this.actorRole,
    this.actorId,
    required this.details,
    this.metadataJson,
    required this.timestamp,
  });

  factory AuditLogModel.fromJson(Map<String, dynamic> json) {
    return AuditLogModel(
      id: json['id'] ?? '',
      supportTicketId: json['supportTicketId'],
      action: json['action'] ?? '',
      actorRole: json['actorRole'] ?? '',
      actorId: json['actorId'],
      details: json['details'] ?? '',
      metadataJson: json['metadataJson'],
      timestamp: DateTime.tryParse(json['timestamp'] ?? '') ?? DateTime.now(),
    );
  }
}

class SupportTicketModel {
  final String id;
  final String userId;
  final String? userName;
  final String? userEmail;
  final String? bookingId;
  final String? tourId;
  final String title;
  final String description;
  final String category;
  final String priority;
  final String status;
  final String? attachmentUrl;
  final double sentimentScore;
  final String severityTier;
  final String? aiReasoning;
  final String? resolutionSummary;
  final DateTime createdAt;
  final DateTime updatedAt;
  final List<AuditLogModel> auditLogs;

  SupportTicketModel({
    required this.id,
    required this.userId,
    this.userName,
    this.userEmail,
    this.bookingId,
    this.tourId,
    required this.title,
    required this.description,
    required this.category,
    required this.priority,
    required this.status,
    this.attachmentUrl,
    required this.sentimentScore,
    required this.severityTier,
    this.aiReasoning,
    this.resolutionSummary,
    required this.createdAt,
    required this.updatedAt,
    this.auditLogs = const [],
  });

  factory SupportTicketModel.fromJson(Map<String, dynamic> json) {
    return SupportTicketModel(
      id: json['id'] ?? '',
      userId: json['userId'] ?? '',
      userName: json['userName'],
      userEmail: json['userEmail'],
      bookingId: json['bookingId'],
      tourId: json['tourId'],
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      category: json['category'] ?? 'General',
      priority: json['priority'] ?? 'Medium',
      status: json['status'] ?? 'Pending_AI_Triage',
      attachmentUrl: json['attachmentUrl'],
      sentimentScore: (json['sentimentScore'] as num?)?.toDouble() ?? 0.0,
      severityTier: json['severityTier'] ?? 'Tier_1_Low',
      aiReasoning: json['aiReasoning'],
      resolutionSummary: json['resolutionSummary'],
      createdAt: DateTime.tryParse(json['createdAt'] ?? '') ?? DateTime.now(),
      updatedAt: DateTime.tryParse(json['updatedAt'] ?? '') ?? DateTime.now(),
      auditLogs: (json['auditLogs'] as List<dynamic>?)
              ?.map((e) => AuditLogModel.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
    );
  }
}
